### DISCUSSION TITLE
Sitemap issue when compiling sass with custom template language example

### DISCUSSION THREAD

**Discussion by bobmonsour** [11ty/eleventy#2850] [Q&A] [answered]
When I add sass as a template language, I get the correct single main.css file in my output dir, but my sitemap includes what appear to be intermediate files produced by the rendering. Here's the code I'm using in my eleventy config to add sass:

```
// Recognize Sass as a "template languages"
  eleventyConfig.addTemplateFormats("scss");

  // Compile Sass
  eleventyConfig.addExtension("scss", {
    outputFileExtension: "css",
    compile: async function (inputContent, inputPath) {
      // Skip files like _fileName.scss
      let parsed = path.parse(inputPath);
      if (parsed.name.startsWith("_")) {
        return;
      }

      // Run file content through Sass
      let result = sass.compileString(inputContent, {
        loadPaths: [parsed.dir || "."],
        sourceMap: false, // or true, your choice!
      });

      // Allow included files from @use or @import to
      // trigger rebuilds when using --incremental
      this.addDependencies(inputPath, result.loadedUrls);

      return async () => {
        return result.css;
      };
    },
  });
```

I have 5 files in my css folder within my input directory:

```
/content (input dir)
  /css
      /_index.scss
      /_message-box.scss
      /_prism-diff.scss
      /_prism-okaidia.css
      /main.scss
```

main.scss includes 4 @ use statements to bring in the other 4 files.

In the resulting sitemap.xml file, in addition to the expected urls, I also get the following:

```
	<url>
		<loc>https://example.com/css/_index.css</loc>
		<lastmod>2023-03-02</lastmod>
	</url>
	
	<url>
		<loc>https://example.com/css/_message-box.css</loc>
		<lastmod>2023-03-02</lastmod>
	</url>
	
	<url>
		<loc>https://example.com/css/_prism-diff.css</loc>
		<lastmod>2023-03-02</lastmod>
	</url>
		
	<url>
		<loc>https://example.com/css/main.css</loc>
		<lastmod>2023-03-08</lastmod>
	</url>
```

It seems that all of the files get added to collections.all.

I have tried to use compileOptions: false, by adding the following just after the outputFileExtension.

```
    compileOptions: {
      permalink: false,
    },
```

I get the following error regarding the sitemap template when I run it with this.

```
[11ty] 1. Having trouble rendering njk template ./content/sitemap/sitemap.xml.njk (via TemplateContentRenderError)
[11ty] 2. (./content/sitemap/sitemap.xml.njk)
[11ty]   TypeError: url.startsWith is not a function (via Template render error)
[11ty] 
[11ty] Original error stack trace: Template render error: (./content/sitemap/sitemap.xml.njk)
[11ty]   TypeError: url.startsWith is not a function
```

 I looked at the other discussion topics that make use of compileOptions, but they seem to be different use cases.

What am I missing here? Any help is appreciated.

I did all this based on the most recent version of the eleventy-base-blog. I eliminated the use of the css bundling and placed the css files in the input dir for this example. Here's a [link to the repo](https://github.com/bobmonsour/eleventy-base-blog-main).

**Comment by 5t3ph** [comment, marked as answer]
Hi @bobmonsour / @zachleat - a temporary workaround is to use a directory data file within /css/ (or other custom template language directory) to block the files from collections. Alternatively, make a custom collection vs. use the `all`.

(PS - Bob, I just discovered this issue as well on Friday, I am pushing a README update to my sass plugin to callout this need for now!) 

```js
// css/css.json
{
  "eleventyExcludeFromCollections": true
}
```

But I also absolutely upvote the need to block this earlier, thanks for filing!

**Reply by bobmonsour** [reply]
Thanks @5t3ph. A simple workaround. Would you consider this an 11ty bug and if so, should I write it up on the issues section?

**Reply by darthmall** [reply]
This doesn't really seem like a bug to me. The assumption is—as far as I know—that templates are site content (markdown, nunjucks, WebC, etc.). It makes sense that custom template types are also automatically added to `collections.all` (and thus end up in your sitemap). Having to opt out in a case where you piggyback on custom templates to process non-content assets seems perfectly reasonable to me.

It could be worth adding this to the docs somewhere as a tip or common pitfall, since I'm sure it will come up a lot.

**Reply by bobmonsour** [reply]
I see your point, but it would seem that the opting out would be nice to include in the custom template engine code, rather than requiring a css.json-like technique. It looks like this is something Zach is considering in issue [#2854](https://github.com/11ty/eleventy/issues/2854).

**Reply by darthmall** [reply]
Oh yeah, that’d be a nice feature.
